### PR TITLE
VL_Refactor-UIcon-tests_Dmytro-Holdobin

### PR DESCRIPTION
**Pull request by Explicit12**
Fixed and refactored UIcon component tests

### CODE REVIEW

File: src/ui.image-icon/tests/UIcon.test.ts [hunk 1]
```diff
@@ -1,71 +1,43 @@
...
 
-      expect(component.attributes("data-name")).toBe(name);
-    });
+      await flushPromises();
 
-    // Edge case, no src and name
-    it("doesn't render when neither name nor src is provided", () => {
-      // Vue adds a comment node as a placeholder when using v-if
-      const expectedHTML = "<!--v-if-->";
+      component.get("[data-test='test']");
```
Comment (itsJohnnyGrid) [inline src/ui.image-icon/tests/UIcon.test.ts]: where is expect?

File: src/ui.image-icon/tests/UIcon.test.ts [hunk 2]
```diff
@@ -1,71 +1,43 @@
...
-        [Symbol.toStringTag]: "Module",
-      };
+    it("Name – renders icon based on provided name", async () => {
+      const name = "check";
 
       const component = mount(UIcon, {
         props: {
-          src: mockComponent as unknown as Props["src"],
+          name,
+          dataTest: "test",
```
Comment (itsJohnnyGrid) [inline src/ui.image-icon/tests/UIcon.test.ts]: why data test is needed? You can use `vl-key` attribute instead

File: src/ui.image-icon/tests/UIcon.test.ts (hunk 1 above)
Comment (Explicit12) [reply src/ui.image-icon/tests/UIcon.test.ts]: You don't need to call expect here, if an element with this data attribute doesn't exist, it will fail.

File: src/ui.image-icon/tests/UIcon.test.ts (hunk 2 above)
Comment (Explicit12) [reply src/ui.image-icon/tests/UIcon.test.ts]: It's left over from my previous solution, we don't need this anymore.

File: src/ui.image-icon/tests/UIcon.test.ts [hunk 3]
```diff
@@ -192,43 +162,64 @@ describe("UIcon.vue", () => {
...
-    // iconRef
-    it("exposes iconRef", () => {
-      const name = "check";
-
+    it("Icon – exposes iconRef and it references the correct SVG element", async () => {
       const component = mount(UIcon, {
         props: {
-          name,
+          name: "check",
+          dataTest: "test",
```
Comment (itsJohnnyGrid) [inline src/ui.image-icon/tests/UIcon.test.ts]: dataTest not in use in the test case code
